### PR TITLE
RBAC Test Improvements

### PR DESCRIPTION
**Pull request by tahmidefaz**

### CODE REVIEW

**Comment by tahmidefaz** [comment]
This is ready for review now. :arrow_up:

File: tests/test_api_facts.py [hunk 1]
```diff
@@ -158,6 +158,8 @@ def test_put_facts_with_RBAC_denied(subtests, mocker, api_put, db_create_host, e
 
             assert_response_status(response_status, 403)
 
+            assert db_get_host(host.id).facts != DB_NEW_FACTS
```
Comment (jharting) [inline tests/test_api_facts.py:161]: these would never be equal as on the left side you have all the namespaces whereas on the right side you have data in DB_FACTS_NAMESPACE only

```suggestion
            assert db_get_host(host.id).facts[DB_FACTS_NAMESPACE]["facts"] != DB_NEW_FACTS
```

**Review by jharting — CHANGES_REQUESTED** [review]
Found a problem in one test. Moreover, these two operations are not tested at all:
* get_host_system_profile_by_id
* get_host_tag_count

File: tests/test_api_facts.py (hunk 1 above)
Comment (tahmidefaz) [reply tests/test_api_facts.py]: Looks like we weren't using the `api_put` correctly either. Fixed this + the `api_put`. :+1:

**Comment by tahmidefaz** [comment]
Just added tests for `get_host_tag_count` and `get_host_system_profile_by_id`. This one needs another round of review now.

**Review by jharting — APPROVED** [review]

**Pull request merged**